### PR TITLE
Remove duplication in History and Alarms

### PR DESCRIPTION
**Pull request by rlqualls**
Duplicate code in History and Alarms was factored out
into a new class named RecordContainer. Now, Alarms is empty
but kept for semantic reasons for the time being.

### CODE REVIEW

**Comment by jgorset** [comment]
Thanks, @rlqualls!

Ideally I would like to refactor `RecordContainer` into a module `Queryable` and include it because it makes more sense semantically, seeing as how it only really works with resources that have an operation `query`. We should also consider deriving its arguments from the resource itself (now I'm just assuming that it accepts `start` and `end` because I already know that's how alarm and history work).

**Comment by rlqualls** [comment]
I actually initially went through 3 phases during this refactor:  
1. Module called Queryable with over-engineered metaprogramming that performed assignments to instance variable based on the included class. i.e History's methods would apply to `@history`, Alarms to `@alarms`.  
2. More sensible Queryable with `@history` and `@alarms` renamed to `@records`.  
3. Unsure whether modules should have `#initialize`, changed to class `RecordContainer`.

TLDR: initially went with Queryable but then left out duplicate initialize method because I wasn't sure whether that was good form or if it would even work. 

oh - just tested and it would work.

```
module Queryable
  def initialize
    @foo = 1
  end

  def foo
    @foo
  end
end

class History
  include Queryable
end

history = History.new
puts history.foo
```

**Comment by jgorset** [comment]
Oh, cool! I'm not sure whether using `initialize` in modules is poor form, either — what say you, @timkurvers and @sindrenm?

**Comment by timkurvers** [comment]
Only thing I can think of would be to allow setting `Queryable#source`, and letting records be lazily loaded:

``` ruby
module Queryable

  attr_accessor :source

  def records
    @records ||= OBIX.parse source
  end

  ...

end
```

``` ruby
class History
  include Queryable

  def initialize(source)
    self.source = source
  end

  ...

end
```

That might open up a whole host of other issues.

**Comment by sindrenm** [comment]
I'm not too fond of defining `initialize` in modules that are to be included, as this would require the including class to redefine (and call `super` inside it) `initialize` as well, as to not completely override the included `initialize` method. Something like @timkurvers' solution above would be more welcome, in my eyes. :smiley:
